src/jquery.multi-select.js

```diff
@@ -100,55 +130,91 @@
         if (key === escapeKey) {
           _this.menuHide();
         }
-      });
+      })
+      .appendTo(this.$container);
 
-      this.$menu.on('change.multiselect', function() {
-        _this.updateButtonContents();
-      });
+      this.constructMenuItems();
+
+      if ( this.settings.presets ) {
+        this.constructPresets();
+      }
+    },
+
+    constructMenuItems: function() {
```

Comment:
Is a separate constructMenuItems function needed, it looks like it's only called once from inside constructMenu.